contracts/Insurance.sol

```diff
@@ -121,10 +123,32 @@ contract Insurance is IInsurance, Ownable, SafetyWithdraw {
      * @dev Withdraws from tracer, and adds amount to the pool's amount field.
      */
     function updatePoolAmount() public override {
-        int256 quote = (tracer.getBalance(address(this))).position.quote;
-        if (quote > 0) {
-            tracer.withdraw(uint256(quote));
-            collateralAmount = collateralAmount + uint256(quote);
+        uint256 quote =
+            uint256((tracer.getBalance(address(this))).position.quote);
+
+        tracer.withdraw(quote);
+
+        if (publicCollateralAmount > 0) {
+            // Amount to pay to public is the ratio of public collateral amount to total funds
+            uint256 payToCollateral =
+                PRBMathUD60x18.mul(
+                    quote,
+                    PRBMathUD60x18.div(
+                        publicCollateralAmount,
+                        publicCollateralAmount + bufferCollateralAmount
+                    )
+                );
+
+            publicCollateralAmount = publicCollateralAmount + payToCollateral;
+
+            // Amount to pay to buffer is the remainder
+            bufferCollateralAmount =
+                bufferCollateralAmount +
+                quote -
+                payToCollateral;
+        } else {
+            // Pay to buffer if nothing in public insurance
+            bufferCollateralAmount = bufferCollateralAmount + quote;
```

Comment:
This doesn't look right. At the time a user withdraws from the public fund, the withdrawal fee goes entirely to the insurance buffer account.

Reply:
This is for updating the amount of buffer and public collateral tokens in the insurance pool. In order to optimise for gas, all insurance funding payments are made in the `TracerPerpetualSwaps` contract; when you actually need to deposit or withdraw from the insurance pool, you need access to those collateral tokens, so `updatePoolAmount` gets them from the perp swaps contract (which is why `tracer.withdraw(quote);` is called at the beginning). 

```
            // Pay to buffer if nothing in public insurance
```
is saying that if no one has deposited collateral into the insurance account, then any amount that the Tracer has paid out to the insurance contract (from insurance funding rate payments) belongs to the buffer account. If someone has (i.e. the if-statement branch, where the collateral in the insurance contract is greater than 0), you pay proportionally to the buffer and public account.

The withdraw fee is still under discussion so I decided to leave it out of scope for this PR.